defaults/cmake/Key4hepConfig.cmake

```
@@ -0,0 +1,48 @@
+macro(key4hep_set_compiler_flags)
+  if (DEFINED KEY4HEP_SET_COMPILER_FLAGS AND NOT KEY4HEP_SET_COMPILER_FLAGS)
+    return()
+  endif()
+
+  set(COMPILER_FLAGS "-fPIC -Wall -Wextra -Wpedantic -Wshadow -Wdeprecated")
+
+  if(CMAKE_CXX_COMPILER_ID MATCHES "^(Apple)?Clang$")
+    set(COMPILER_FLAGS "${COMPILER_FLAGS} -Winconsistent-missing-override -Wheader-hygiene -fcolor-diagnostics")
+  elseif(CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
+    set(COMPILER_FLAGS "${COMPILER_FLAGS} -fdiagnostics-color=always")
+  endif()
+
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${COMPILER_FLAGS}")
+
+endmacro()
+
+macro(key4hep_set_build_type)
+  message(WARNING "CMAKE_CONFIGURATION_TYPES: ${CMAKE_CONFIGURATION_TYPES}")
+  if(NOT CMAKE_CONFIGURATION_TYPES)
+    if(NOT CMAKE_BUILD_TYPE)
+      set(CMAKE_BUILD_TYPE RelWithDebInfo
+        CACHE STRING "Choose the type of build, options are: None Release MinSizeRel Debug RelWithDebInfo"
+        FORCE
+        )
+    else()
+      set(CMAKE_BUILD_TYPE "${CMAKE_BUILD_TYPE}"
+        CACHE STRING "Choose the type of build, options are: None Release MinSizeRel Debug RelWithDebInfo"
+        FORCE
+        )
+    endif()
+  endif()
+endmacro()
+
+macro(key4hep_set_cxx_standard)
+  set(CMAKE_CXX_STANDARD 20 CACHE STRING "")
+
+  if(NOT CMAKE_CXX_STANDARD MATCHES "20|23")
+    message(FATAL_ERROR "Unsupported C++ standard: ${CMAKE_CXX_STANDARD}, supported values are 20 and 23")
+  endif()
+
```

```suggestion
set(CMAKE_CXX_STANDARD_REQUIRED ON)
set(CMAKE_CXX_EXTENSIONS OFF)
```
I think these would fit here as well to make it a bit harder to do "stupid things".

---

Yeah, the `CMAKE_CXX_STANDARD_REQUIRED` might already be handled by using an appropriate `CMAKE_CXX_STANDARD`. However, I think `CMAKE_CXX_EXTENSIONS=OFF` would help us from accidentally picking up things that are only supported by one compiler earlier than otherwise potentially. My guess is that at the moment it doesn't really matter, but it is usually easier to lift some restrictions rather then imposing them later.